### PR TITLE
Registered user can become subscriber

### PR DESCRIPTION
**Pull request by swardbladpaulo**
[User can buy a subscription ](https://www.pivotaltracker.com/story/show/176343730)
### User Story
```
As a user
In order to get access to the best articles
I would like to be able to buy a subscription
```
## Changes proposed in this pull request:
* Create a `PaymentForm` component and add it to the `MasterModal`
* Create functionality for paying with stripe using a token
## What I have learned working on this feature:
* the effectiveness of stripe hooks!

## Packages/Gems added
* `@stripe/stripe-js`
* `@stripe/react-stripe-js`
## Screenshots (Client)
Screenshot 2021-01-16 at 11 26 01
Screenshot 2021-01-16 at 11 25 30

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -6,17 +6,16 @@ import RegistrationForm from "./components/RegistrationForm";
...
-  );
+	return (
+		<>
+			<MenuHeader />
+			<Switch>
+				<Route exact path="/" component={DisplayArticlesData}></Route>
+				<Route exact path="/articles/:id" component={SingleArticle}></Route>
+				<Route exact path="/sign_up" component={RegistrationForm}></Route>
+			</Switch>
+		</>
```
Comment (FlisAnn) [inline src/App.jsx]: Is the formatting off?

File: src/components/MasterModal.jsx [hunk 2]
```diff
@@ -1,82 +1,101 @@
...
 import { Button, Icon, Modal, Message } from "semantic-ui-react";
 import RegistrationForm from "./RegistrationForm";
+import PaymentForm from "./PaymentForm";
 import { performAuthentication } from "../modules/auth";
 import { useDispatch, useSelector } from "react-redux";
+import { loadStripe } from "@stripe/stripe-js";
+import { Elements } from "@stripe/react-stripe-js";
+
+const stripePromise = loadStripe(
+	"pk_test_51HuxhoEDdj3L9cb7mRwXVf3mjnVY7dk7NL7WmIZ31HGjkKHk9RSQASdCWIXiObHTiyfYkNhxKFsvLFkvLKyxM6Wz004BVP7pWj"
```
Comment (FlisAnn) [inline src/components/MasterModal.jsx]: Consider whether this key should be here

File: src/components/MasterModal.jsx [hunk 3]
```diff
@@ -1,82 +1,101 @@
...
-            form="registrationForm"
-            data-cy="submit-btn"
-            primary
-          >
-            Proceed to Payment <Icon name="right chevron" />
-          </Button>
-        </Modal.Actions>
+	return (
+		<>
+			<Button onClick={() => setFirstOpen(true)} data-cy="register-btn">
```
Comment (FlisAnn) [inline src/components/MasterModal.jsx]: Is the formatting off here?

File: src/index.js [hunk 4]
```diff
@@ -1,25 +1,24 @@
...
-    </BrowserRouter>
-  </Provider>,
-  document.getElementById("root")
+	<Provider store={store}>
+		<BrowserRouter>
+			<App />
+		</BrowserRouter>
+	</Provider>,
+	document.getElementById("root")
 );
```
Comment (FlisAnn) [inline src/index.js]: ist the formatting off?

File: src/modules/articlesData.js [hunk 5]
```diff
@@ -2,28 +2,28 @@ import axios from "axios";
...
-        payload: response.data.article,
-      });
-    } catch (error) {
-      store.dispatch({
-        type: "SET_ERROR_MESSAGE",
-        payload: error.response.data.message,
-      });
-    }
-  },
+	async show(articleId) {
```
Comment (FlisAnn) [inline src/modules/articlesData.js]: formatting?

File: src/modules/auth.js [hunk 6]
```diff
@@ -1,33 +1,33 @@
...
+				authenticated: isAuthenticated,
+				currentUser: response.data.data,
+			},
+		});
+	} catch (error) {
+		dispatch({
+			type: "SET_ERROR_MESSAGE",
+			payload: error.response.data.errors,
+		});
+	}
```
Comment (FlisAnn) [inline src/modules/auth.js]: formatting?

File: src/state/reducers/rootReducer.js [hunk 7]
```diff
@@ -1,33 +1,42 @@
...
+					role: action.payload.role,
+				},
+			};
+
+		case "SET_ERROR_MESSAGE":
+			return {
+				...state,
+				errorMessage: action.payload,
+			};
+		default:
```
Comment (FlisAnn) [inline src/state/reducers/rootReducer.js:37]: formatting?

**Review by FlisAnn — COMMENTED** [review]
Please have a look at the formatting if something is off?

**Review by rithreaksa — COMMENTED** [review]
Love your naming standard and consistency. Good job.  Pls just check the formatting.

File: src/state/reducers/rootReducer.js (hunk 7 above)
Comment (sealfury) [reply src/state/reducers/rootReducer.js:37]: Have fixed formatter settings and pushed it all back up again, sorry about that! 😳

**Comment by FlisAnn** [comment]
Merging since formatting is done

**Pull request closed without merging**